code/core/system/proxy-velodyne64/testsuites/ProxyVelodyne64TestSuite.h

```diff
@@ -22,21 +22,202 @@
+            }
+        }
+    }
+
+   private:
+    MyContainerConference mcc;
+    opendlv::core::system::proxy::velodyne64Decoder v64decoder;
+};
 
 class ProxyVelodyne64Test : public CxxTest::TestSuite {
    public:
-    void setUp() {}
+    ProxyVelodyne64Test()
+        : velodyneSM(SharedMemoryFactory::createSharedMemory(NAME, SIZE))
+        , segment(SharedMemoryFactory::createSharedMemory(NAME2, SIZE)) {}
+
+    ~ProxyVelodyne64Test() {}
+
+    void readCsvFile() {                                              //Read a .csv file that is compressed into a .zip file
+        fstream fin("../atwallFrame1New.zip", ios::binary | ios::in); //Read xyz and intensity values of points in Frame 1 exported from VeloView. These values exist in a zip file in the parent folder
+        if (!fin.is_open()){
+            cout << "The file containing VeloView reference data is not found." << endl;
+        }
+        std::shared_ptr< odcore::wrapper::DecompressedData > dd = odcore::wrapper::CompressionFactory::getContents(fin);
+        fin.close();
+        vector< string > entries = dd->getListOfEntries();
+        std::shared_ptr< istream > stream = dd->getInputStreamFor(entries.at(0));
+        stringstream decompressedData; //Store the decompressed data from the zip file
+        if (stream.get()) {
+            char c;
+            while (stream->get(c)) {
+                decompressedData << c;
+            }
+        }
+        string value;
+        getline(decompressedData, value);          //The .csv file after decompression has this format: "x, y, z, intensity,", while each line represents one point
+        while (getline(decompressedData, value)) { //Read the csv file line by line and save x, y, z, intensity values of all points in Frame 1 to respective vectors for comparison with our Velodyne64 decoder
+            stringstream lineStream(value);
+            string cell;
+            getline(lineStream, cell, ',');
+            xDataV.push_back(stof(cell));
+            getline(lineStream, cell, ',');
+            yDataV.push_back(stof(cell));
+            getline(lineStream, cell, ',');
+            zDataV.push_back(stof(cell));
+            getline(lineStream, cell, ',');
+            intensityV.push_back(stof(cell));
+        }
+    }
+
+
+    void testVelodyneDecodingFromFile() {
+        readCsvFile();
+        PCAPProtocol pcap;                     //Use the PCAP decoder of OpenDaVINCI to read the .pcap recording
+        packetToByte p2b(velodyneSM, segment); //This class extracts Velodyne payload from pcap packets
+        pcap.setContainerListener(&p2b);       //Set the packetToByte class as the container listener of the PCAP decoder
 
-    void tearDown() {}
+        fstream lidarStream("../atwallshort.pcap", ios::binary | ios::in); //A sample .pcap file containing several Velodyne frames in the parent folder
+        if (!lidarStream.is_open()){
```

Comment:
As we are in a test case, just replace the if(...) {...} by TS_ASSERT(lidarStream.is_open()); to make the test case fail otherwise.